compat/mingw.c

```diff
@@ -2303,7 +2303,7 @@ int symlink(const char *target, const char *link)
 	return 0;
 }
 
-#ifndef _WINNT_H
+#if !defined(_WINNT_H) && !defined(_MSC_VER)
```

Comment:
Oh, and: would it not be better to simply guard the `_ANONYMOUS_UNION` in an `#ifndef _MSC_VER ... #endif`?